crates/line_ending/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+//! We maintain the invariant that all internal strings use `\n` as line separator.
+//! This module does line ending conversion and detection (so that we can
+//! convert back to `\r\n` on the way out as needed).
+
+use std::sync::LazyLock;
+
+use memchr::memmem;
+
+static FINDER: LazyLock<memmem::Finder> = LazyLock::new(|| memmem::Finder::new(b"\r\n"));
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum LineEnding {
+    ///  Line Feed only (\n), common on Linux and macOS as well as inside git repos
+    Lf,
+
+    /// Carriage Return + Line Feed characters (\r\n), common on Windows
+    Crlf,
```

Comment:
I switched to the convention that `biome_formatter::LineEnding` uses so it maps over nicely